src/hierarchies/lapnig_hierarchy.cc

```diff
@@ -0,0 +1,209 @@
+#include "hierarchy_prior.pb.h"
+#include "ls_state.pb.h"
+#include "src/utils/rng.h"
+
+unsigned int LapNIGHierarchy::accepted_ = 0;
+unsigned int LapNIGHierarchy::iter_ = 0;
+
+
+void LapNIGHierarchy::set_state_from_proto(
+    const google::protobuf::Message &state_) {
+  auto &statecast = downcast_state(state_);
+  state.mean = statecast.uni_ls_state().mean();
+  state.scale = statecast.uni_ls_state().var(); // ??? changed from scale to var
+  set_card(statecast.cardinality());
+}
+
+void LapNIGHierarchy::set_hypers_from_proto(
+    const google::protobuf::Message &hypers_) {
+  auto &hyperscast = downcast_hypers(hypers_).lapnig_state();
+  hypers->mean = hyperscast.mean();
+  hypers->var = hyperscast.var();
+  hypers->scale = hyperscast.scale();
+  hypers->shape = hyperscast.shape();
+}
+
+double LapNIGHierarchy::like_lpdf(const Eigen::RowVectorXd &datum) const {
+  return stan::math::double_exponential_lpdf(datum(0), state.mean,
+                                             state.scale);
+}
+
+std::shared_ptr<bayesmix::AlgorithmState::ClusterState>
+LapNIGHierarchy::get_state_proto() const {
+  bayesmix::UniLSState state_;
+  state_.set_mean(state.mean);
+  state_.set_var(state.scale);
+
+  auto out = std::make_shared<bayesmix::AlgorithmState::ClusterState>();
+  out->mutable_uni_ls_state()->CopyFrom(state_);
+  return out;
+}
+
+std::shared_ptr<bayesmix::AlgorithmState::HierarchyHypers>
+LapNIGHierarchy::get_hypers_proto() const {
+  bayesmix::LapNIGState hypers_;
+  hypers_.set_mean(hypers->mean);
+  hypers_.set_var(hypers->var);
+  hypers_.set_shape(hypers->shape);
+  hypers_.set_scale(hypers->scale);
+
+  auto out = std::make_shared<bayesmix::AlgorithmState::HierarchyHypers>();
+  out->mutable_lapnig_state()->CopyFrom(hypers_);
+  return out;
+}
+
+void LapNIGHierarchy::clear_summary_statistics() {
+  cluster_data_values.clear();
+}
+
+void LapNIGHierarchy::initialize_hypers() {
+  if (prior->has_fixed_values()) {
+    // Set values
+    hypers->mean = prior->fixed_values().mean();
+    hypers->var = prior->fixed_values().var();
+    hypers->shape = prior->fixed_values().shape();
+    hypers->scale = prior->fixed_values().scale();
+    // Check validity
+    if (hypers->var <= 0) {
+      throw std::invalid_argument("Variance-scaling parameter must be > 0");
+    }
+    if (hypers->shape <= 0) {
+      throw std::invalid_argument("Shape parameter must be > 0");
+    }
+    if (hypers->scale <= 0) {
+      throw std::invalid_argument("Scale parameter must be > 0");
+    }
+  }
+  else {
+    throw std::invalid_argument("Unrecognized hierarchy prior ghjk");
+  }
+}
+
+void LapNIGHierarchy::initialize_state() {
+  state.mean = hypers->mean;
+  state.scale = 1./hypers->scale / (hypers->shape + 1);
+}
+
+void LapNIGHierarchy::update_hypers(
+    const std::vector<bayesmix::AlgorithmState::ClusterState> &states) {
+  auto &rng = bayesmix::Rng::Instance().get();
+
+  if (prior->has_fixed_values()) {
+    return;
+  }
```

Comment:
So the convention is basically to have else clause on the same line if there is just one if clause? If yes, then what if the if clause is long?
We noticed some more cases when we haven't followed that convention, if you confirm we'll change also in those cases.

Reply:
You do not need to update style choices by hand, you can do so automatically. Please check out [CONSTRIBUTORS.md](https://github.com/bayesmix-dev/bayesmix/blob/master/CONTRIBUTORS.md) for more information. Please make sure to run one of the options every now and then, so that we can keep an uniform style across the library.

Reply:
Cool, thanks!